model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -297,10 +317,22 @@ public boolean removeMember(OrganizationModel organization, UserModel member) {
         if (isManagedMember(organization, member)) {
             userProvider.removeUser(realm, member);
         } else {
-            List<String> organizations = member.getAttributes().get(ORGANIZATION_ATTRIBUTE);
-            organizations.remove(organization.getId());
-            member.setAttribute(ORGANIZATION_ATTRIBUTE, organizations);
-            member.leaveGroup(getOrganizationGroup(organization));
+            OrganizationModel current = (OrganizationModel) session.getAttribute(OrganizationModel.class.getName());
+
+            if (current == null) {
```

Comment:
This check is because we don't want to remove the attribute if an organization has been previously set.